src/Mado/QueryBundle/Queries/AndFilterFactory.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace Mado\QueryBundle\Queries;
+
+use Mado\QueryBundle\Services\StringParser;
+
+class AndFilterFactory
+{
+    private $entityAlias;
+
+    private $fields;
+
+    private $conditions;
+
+    private $parameters;
+
+    private $relationEntityAlias;
+
+    private $parser;
+
+    private $joinFactory;
+
+    public function __construct(string $entityAlias, array $fields, JoinFactory $joinFactory)
+    {
+        $this->entityAlias = $entityAlias;
+        $this->fields = $fields;
```

Comment:
Could I suggest you to keep in constructor initializations same order of method proprtype?